tests/std/tests/P1208R6_source_location/test.cpp

```diff
@@ -87,28 +105,53 @@ constexpr void sub_member_test() {
     assert(s.x.loc.line() == __LINE__ - 1);
     assert(s.x.loc.column() == 14);
     if (is_constant_evaluated()) {
-        assert(s.x.loc.function_name() == "main"sv); // TRANSITION, VSO-1285783
+        assert(s.x.loc.function_name() == "int __cdecl main(void)"sv); // TRANSITION, VSO-1285783
     } else {
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-10199227 and LLVM-58951
         assert(s.x.loc.function_name() == "sub_member_test"sv);
+#else // ^^^ workaround / no workaround vvv
+        assert(s.x.loc.function_name() == "void __cdecl sub_member_test(void)"sv);
+#endif // TRANSITION, DevCom-10199227 and LLVM-58951
     }
     assert(string_view{s.x.loc.file_name()}.ends_with(test_cpp));
 
     const s2 s_i{1};
     assert(s_i.x.loc.line() == s2_int_line);
     assert(s_i.x.loc.column() == 5);
+#if defined(__clang__) || defined(__EDG__) // TRANSITION, DevCom-10199227 and LLVM-58951
     assert(s_i.x.loc.function_name() == "s2"sv);
+#elif defined(_M_IX86) // ^^^ workaround / no workaround vvv
+    assert(s_i.x.loc.function_name() == "__thiscall s2::s2(int)"sv);
+#else // ^^^ _M_IX86 / !_M_IX86 vvv
+    assert(s_i.x.loc.function_name() == "__cdecl s2::s2(int)"sv);
+#endif // TRANSITION, DevCom-10199227 and LLVM-58951
     assert(string_view{s_i.x.loc.file_name()}.ends_with(test_cpp));
 }
 
 constexpr void lambda_test() {
-    const auto l = [loc = source_location::current()] { return loc; };
-    const auto x = l();
-    assert(x.line() == __LINE__ - 2);
+    const auto x1 = [loc = source_location::current()] { return loc; }();
+    const auto x2 = [] { return source_location::current(); }();
+    assert(x1.line() == __LINE__ - 2);
+    assert(x2.line() == __LINE__ - 2);

```

Comment:
Previously, by defining `l` and `x` separately, this verified that the `source_location` cared about the point of definition, not the callsite. That minor coverage appears to be lost here.